src/geo_traits_impl/line_string.rs

```diff
@@ -0,0 +1,59 @@
+use super::{LineStringType, PointType};
+use bytemuck::TransparentWrapper;
+use geo_traits::{Dimensions, LineStringTrait, PointTrait};
+
+impl LineStringTrait for LineStringType {
+    type T = f64;
+    type CoordType<'b>
+        = &'b PointType
+    where
+        Self: 'b;
+
+    fn num_coords(&self) -> usize {
+        self.0.len()
+    }
+
+    fn coord(&self, i: usize) -> Option<Self::CoordType<'_>> {
+        Some(PointType::wrap_ref(&self.0[i]))
+    }
+
+    unsafe fn coord_unchecked(&self, i: usize) -> Self::CoordType<'_> {
+        self.coord(i).unwrap()
+    }
+
+    fn coords(&self) -> impl DoubleEndedIterator + ExactSizeIterator<Item = Self::CoordType<'_>> {
+        self.0.iter().map(PointType::wrap_ref)
+    }
```

Comment:
Similarly, is there a benefit to overriding the default iterator?

Reply:
Nope! I didn't realize it had a default implementation

Reply:
https://github.com/georust/geojson/pull/245/commits/40bec4b3efe859e7978113465b52921cae6a40fe